src/lv_misc/lv_txt_ap.c

```diff
@@ -0,0 +1,263 @@
+  {1  , 0xFE84, -1, 0, -1,  {1, 0}},  // أ
+  {2  , 0xFE86, -1, 0, -1,  {1, 0}},  // ؤ
+  {3  , 0xFE88, -1, 0, -1,  {1, 0}},  // ﺇ
+  {4  , 0xFE8A, 1, 2, -1,  {1, 0}},  // ئ
+  {5  , 0xFE8E, -1, 0, -1,  {1, 0}},  // آ
+  {6  , 0xFE90, 1, 2, -1,  {1, 1}},  // ب
+  {92 , 0xFB57, 1, 2, -1,  {1, 1}},  // پ
+  {8  , 0xFE96, 1, 2, -1,  {1, 1}},  // ت
+  {9  , 0xFE9A, 1, 2, -1,  {1, 1}},  // ث
+  {10 , 0xFE9E, 1, 2, -1,  {1, 1}},  // ج
+  {100, 0xFB7B, 1, 2, -1,  {1, 1}},  // چ
+  {11 , 0xFEA2, 1, 2, -1,  {1, 1}},  // ح
+  {12 , 0xFEA6, 1, 2, -1,  {1, 1}},  // خ
+  {13 , 0xFEAA, -1, 0, -1,  {1, 0}},  // د
+  {14 , 0xFEAC, -1, 0, -1,  {1, 0}},  // ذ
+  {15 , 0xFEAE, -1, 0, -1,  {1, 0}},  // ر
+  {16 , 0xFEB0, -1, 0, -1,  {1, 0}},  // ز
+  {118, 0xFB8B, -1, 0, -1,  {1, 0}},  // ژ
+  {17 , 0xFEB2, 1, 2, -1,  {1, 1}},  // س
+  {18 , 0xFEB6, 1, 2, -1,  {1, 1}},  // ش
+  {19 , 0xFEBA, 1, 2, -1,  {1, 1}},  // ص
+  {20 , 0xFEBE, 1, 2, -1,  {1, 1}},  // ض
+  {21 , 0xFEC2, 1, 2, -1,  {1, 1}},  // ط
+  {22 , 0xFEC6, 1, 2, -1,  {1, 1}},  // ظ
+  {23 , 0xFECA, 1, 2, -1,  {1, 1}},  // ع
+  {24 , 0xFECE, 1, 2, -1,  {1, 1}},  // غ
+  {31 , 0xFED2, 1, 2, -1,  {1, 1}},  // ف
+  {32 , 0xFED6, 1, 2, -1,  {1, 1}},  // ق
+  {135, 0xFB8F, 1, 2, -1,  {1, 1}},  // ک
+  {33, 0xFEDA, 1, 2, -1,  {1, 1}},  // ﻙ
+  {141, 0xFB93, 1, 2, -1,  {1, 1}},  // گ
+  {34 , 0xFEDE, 1, 2, -1,  {1, 1}},  // ل
+  {35 , 0xFEE2, 1, 2, -1,  {1, 1}},  // م
+  {36 , 0xFEE6, 1, 2, -1,  {1, 1}},  // ن
+  {38 , 0xFEEE, -1, 0, -1,  {1, 0}},  // و
+  {37 , 0xFEEA, 1, 2, -1,  {1, 1}},  // ه
+  {39 , 0xFBFD, 1, 2, -1,  {1, 1}},  // ي
+  {40 , 0xFEF2, 1, 2, -1,  {1, 1}},  // ي
+  {170 , 0xFBFD, 1, 2, -1,  {1, 1}},  // ی
+  {7 , 0xFE94, 1, 2, -1,  {1, 0}},  // ة
+  {206, 0x06F0, 1, 2, -1,  {0, 0}},  // ۰
+  {207, 0x06F1, 0, 0, 0,  {0, 0}},  // ۱
+  {208, 0x06F2, 0, 0, 0,  {0, 0}},  // ۲
+  {209, 0x06F3, 0, 0, 0,  {0, 0}},  // ۳
+  {210, 0x06F4, 0, 0, 0,  {0, 0}},  // ۴
+  {211, 0x06F5, 0, 0, 0,  {0, 0}},  // ۵
+  {212, 0x06F6, 0, 0, 0,  {0, 0}},  // ۶
+  {213, 0x06F7, 0, 0, 0,  {0, 0}},  // ۷
+  {214, 0x06F8, 0, 0, 0,  {0, 0}},  // ۸
+  {215, 0x06F9, 0, 0, 0,  {0, 0}},  // ۹
+  LV_AP_END_CHARS_LIST
+};
+/**********************
+*      MACROS
+**********************/
+
+/**********************
+*   GLOBAL FUNCTIONS
+**********************/
+uint32_t lv_txt_ap_calc_bytes_cnt(const char * txt){
+    uint32_t txt_length = 0;
+    uint32_t chars_cnt = 0;
+    uint32_t current_ap_idx = 0;
+    uint32_t i, j;
+    uint32_t ch_enc;
+
+    txt_length = lv_txt_get_encoded_length(txt);
+
+    i=0;
+    j=0;
+    while(i<txt_length){
+    	ch_enc = lv_txt_encoded_next(txt, &j);
+    	current_ap_idx = lv_ap_get_char_index(ch_enc);
+
+        if(current_ap_idx != LV_UNDEF_ARABIC_PERSIAN_CHARS)
+            ch_enc = ap_chars_map[current_ap_idx].char_end_form;
+
+		if(ch_enc <=0x7F)
+		  chars_cnt++;
+		else if(ch_enc <=0x7FFF)
+		  chars_cnt+=2;
+		else
+		  chars_cnt+=3;
+
+		i++;
```

Comment:
Should this entire block be inside the `if` statement?
The indentation here is confusing because this block of code looks visually as if it's under the `if` block, while it's really under the `while` block...

Reply:
I'm agree, You are right.